### DISCUSSION TITLE
Adding a new source - NetAuth

### DISCUSSION THREAD

**Discussion by the-maldridge** [goauthentik/authentik#11506] [General]
I'm interested in developing a new source connector for [NetAuth](https://github.com/netauth/netauth).  A contributor recently built python bindings for NetAuth's API, so I think there's a decent start there, but its been well over a decade since I last wrote production Python.

In Go, if I wanted to add a new implementation for a widget I could look at the interface definition to figure out what I need to implement.  Does Python and/or Django have anything similar to this?

I have read through the LDAP source and I think this is largely copying information inwards to Authentik's database.  I'd prefer to do lookups on-line where possible, is this supported?

Lastly I really don't expect this is a source that would be accepted by the Authentik project, how hard is it to package up a sub-application for Django and distribute it?  I know its possible to do this from experience with NetBox plugins, but I don't really know the mechanics of how this gets implemented.

TL;DR - I'm looking for pointers as to what I don't know to then ask better questions.  My end goal is to add support for NetAuth as a source.
